modules/xfem/src/base/XFEM.C

```diff
@@ -59,7 +59,7 @@ XFEM::getCrackTipOrigin(std::map<unsigned int, const Elem* > & elem_id_crack_tip
   std::map<unsigned int, const Elem*> elem_id_map;
 
   int m = -1;
-  for (mit1 = _elem_crack_origin_direction_map.begin(); mit1 != _elem_crack_origin_direction_map.end(); mit1++)
+  for (; mit1 != _elem_crack_origin_direction_map.end(); ++mit1)
```

Comment:
I'm not sure why you like to declare these iterators outside of the loop. It's generally a good programming practice to keep variables in the most local scope possible. We make exceptions for performance but this isn't one of those.